### PR TITLE
doc: move dev_model and add contribution guide

### PR DESCRIPTION
**Pull request by greg-fer**
Moved the development model section to top level.
Added contributions guide for the NCS on the top level next to it. NCSDK-23303.

### CODE REVIEW

File: doc/nrf/dev_model_and_contributions/contributions.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Contributions are welcome but entirely optional, as the :ref:`licenses <dm_licenses>` used by the |NCS| allow for free modification of the source code without a requirement to contribute the changes back.
+
+The |NCS| contribution guidelines are modelled on :ref:`those adopted in Zephyr <zephyr:contribute_guidelines>`.
+The following table lists the major contribution guidelines and summarizes the similarities and differences between Zephyr and the |NCS|.
+
+.. list-table:: Contribution guidelines in Zephyr and the |NCS|
+   :widths: auto
+   :header-rows: 1
+
+   * - Contribution guideline in Zephyr
```
Comment (carlescufi) [inline doc/nrf/dev_model_and_contributions/contributions.rst:20]: guidelines

File: doc/nrf/dev_model_and_contributions/contributions.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+     - Description
+     - Differences with the |NCS|
+   * - Check :ref:`Licensing <zephyr:licensing_requirements>`
+     - Apache 2.0 in Zephyr
+     - `LicenseRef-Nordic-5-Clause <LICENSE file_>`_ in the |NCS|.
+   * - Respect :ref:`Copyright notices <zephyr:copyrights>`
+     - Recognize the ownership of the original work according to the Linux Foundation's `Community Best Practice for Copyright Notice`_.
+     - Zephyr guideline valid in the |NCS|.
```
Comment (carlescufi) [inline doc/nrf/dev_model_and_contributions/contributions.rst:28]: guidelines

File: doc/nrf/dev_model_and_contributions/contributions.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+   * - Gain familiarity with source tree structure
+     - In Zephyr, this means respecting the :ref:`zephyr:source_tree_v2` and the purpose of main files and directories in order to avoid redundancy.
+     - While the guideline is valid for the |NCS|, the SDK uses a different :ref:`repository organization <dm_code_base>`.
+   * - Check issues before contributing
+     - When contributing, check the existing pull requests and issues in the respective repository in order to avoid redundancy.
+     - In the |NCS|, verify also the existing :ref:`known_issues`.
+   * - Run local scripts before contributing
+     - Zephyr recommends :ref:`running a series of local scripts <zephyr:Continuous Integration>` to verify the code changes that are to be contributed.
+     - Zephyr guideline valid in the |NCS|.
+   * - Respect coding style
```
Comment (carlescufi) [inline doc/nrf/dev_model_and_contributions/contributions.rst:45]: the coding style

File: doc/nrf/dev_model_and_contributions/contributions.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+     - Zephyr provides a series of :ref:`coding style recomendations <zephyr:coding_style>` and :ref:`zephyr:coding_guidelines`.
+     - Zephyr guideline valid in the |NCS|.
```
Comment (carlescufi) [inline doc/nrf/dev_model_and_contributions/contributions.rst:47]: guidelines

Comment (carlescufi) [reply doc/nrf/dev_model_and_contributions/contributions.rst:47]: also all other occurrences of "guideline" in singular

File: doc/nrf/dev_model_and_contributions/contributions.rst [hunk 5]
```diff
@@ -0,0 +1,56 @@
+.. _contributions_ncs:
+
+Contributing to the |NCS|
+#########################
+
+.. contents::
+   :local:
+   :depth: 2
+
+Any individual or company can choose to contribute to the |NCS| codebase.
```
Comment (carlescufi) [inline doc/nrf/dev_model_and_contributions/contributions.rst:10]: I think we should make it clear here that, if the issue is in Zephyr code, it is much preferrable to contribute it directly upstream to the Zephyr project.

**Review by carlescufi — CHANGES_REQUESTED** [review]

**Review by ihansse — APPROVED** [review]
Look fine to me

File: doc/nrf/dev_model_and_contributions/contributions.rst (hunk 5 above)
Comment (greg-fer) [reply doc/nrf/dev_model_and_contributions/contributions.rst:10]: Added a note.

**Review by carlescufi — APPROVED** [review]

**Pull request merged**